File: dataset/output/OPENAI_gpt-4o/Task19_OPENAI_gpt-4o.java
// Java
public class Main {
    public static void main(String[] args) {
        String input = "Hey fellow warriors";
        System.out.println(reverseWords(input));
    }
    
    public static String reverseWords(String str) {
        String[] words = str.split(" ");
        for (int i = 0; i < words.length; i++) {
            if (words[i].length() >= 5) {
                words[i] = new StringBuilder(words[i]).reverse().toString();
            }
        }
        return String.join(" ", words);
    }
}

File: dataset/output/OPENAI_gpt-4o/Task19_OPENAI_gpt-4o.cpp
#include <iostream>
#include <sstream>
#include <vector>
#include <algorithm>

std::string spinWords(const std::string &sentence) {
    std::istringstream iss(sentence);
    std::ostringstream oss;
    std::string word;
    bool first = true;

    while (iss >> word) {
        if (!first) oss << " ";
        if (word.length() >= 5) {
            std::reverse(word.begin(), word.end());
        }
        oss << word;
        first = false;
    }
    return oss.str();
}

int main() {
    std::vector<std::string> examples = {"Hey fellow warriors", "This is a test", "This is another test"};
    for (const auto& example : examples) {
        std::cout << spinWords(example) << std::endl;
    }
    return 0;
}